File: problem_6/sum_square_difference_test.rb
require 'minitest/autorun'
require 'minitest/pride'

class SumSquareDifferenceTest < Minitest::Test
  def sum_square_difference(max)
    range = (1..max).to_a

    square_of_sums = range.sum ** 2

    sum_of_squares = 0
    range.each { |num| sum_of_squares += num ** 2 }

    square_of_sums - sum_of_squares
  end

  def test_sum_square_difference
    assert_equal 2640, sum_square_difference(10)
    assert_equal 25164150, sum_square_difference(100)
    assert_equal 250166416500, sum_square_difference(1000)
  end
end

File: problem_6/sum-square-difference.test.js
function sumSquareDifference(max) {
  let sum = 0;
  let sumOfSquares = 0;

  for (let i = 1; i <= max; i++) {
    sum += i;
    sumOfSquares += Math.pow(i, 2);
  }

  let squareOfSum = Math.pow(sum, 2);

  return squareOfSum - sumOfSquares;
}

test("sumSquareDifference", () => {
  expect(sumSquareDifference(10)).toBe(2640);
  expect(sumSquareDifference(100)).toBe(25164150);
  expect(sumSquareDifference(1000)).toBe(250166416500);
});
